### PR TITLE
Update ConverterFilterKind enum to be flags

### PR DESCRIPTION
**Pull request by kaiguo**
Idea comes from this API review comment https://github.com/microsoft/microsoft-ui-xaml-specs/pull/48/files#r317833889.

ConverterFilterKind is now a flag, that enables us to combine values like below

```xml
<CornerRadiusFilterConverter Filter="TopLeft,BottomRight" />
```

### CODE REVIEW

File: dev/Common/CornerRadiusFilterConverters.idl [hunk 1]
```diff
@@ -16,15 +16,19 @@ runtimeclass CornerRadiusFilterConverter : Windows.UI.Xaml.DependencyObject, Win
...
+    None = 0x0000,
+    TopLeft = 0x0001,
+    TopRight = 0x0002,
+    BottomLeft = 0x0004,
+    BottomRight = 0x0008,
+    Top = TopLeft | TopRight,
+    Left = TopLeft | BottomLeft,
+    Bottom = BottomLeft | BottomRight,
+    Right = TopRight | BottomRight,
+    ValueOnly = 0x0010,
```
Comment (MikeHillberg) [inline dev/Common/CornerRadiusFilterConverters.idl]: What's the purpose of ValueOnly?

Comment (kaiguo) [reply dev/Common/CornerRadiusFilterConverters.idl]: This is to work around the CX binding bug we saw in the email thread. See #1239.

**Comment by kaiguo** [comment]
@jevansaks Looks like CX doesn't support value combinations like `Filter="TopLeft,ValueOnly"`

Here's the generated code in metadata provider, fromString value is the raw string "TopLeft,ValueOnly", it throws an invalid arg exception. I guess I'll just close this PR and our existing code?

```c++
    XamlMetadataProvider::RegisterXamlType(
        /* Arg1 TypeName */ 
        L"Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusFilterKind",
        /* Arg2 CreateXamlTypeCallback */ 
        []()
        {
            auto xamlType = winrt::make<EnumXamlType>(
                /* Arg 1 - TypeName */ 
                (PCWSTR)L"Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusFilterKind",
                /* Arg 2 - CreateFromString func */ 
                (std::function<winrt::IInspectable(hstring)>)[](hstring fromString)
                {
                    if (fromString == L"None") return box_value(winrt::CornerRadiusFilterKind::None);
                    if (fromString == L"TopLeft") return box_value(winrt::CornerRadiusFilterKind::TopLeft);
                    if (fromString == L"TopRight") return box_value(winrt::CornerRadiusFilterKind::TopRight);
                    if (fromString == L"BottomLeft") return box_value(winrt::CornerRadiusFilterKind::BottomLeft);
                    if (fromString == L"BottomRight") return box_value(winrt::CornerRadiusFilterKind::BottomRight);
                    if (fromString == L"Top") return box_value(winrt::CornerRadiusFilterKind::Top);
                    if (fromString == L"Left") return box_value(winrt::CornerRadiusFilterKind::Left);
                    if (fromString == L"Bottom") return box_value(winrt::CornerRadiusFilterKind::Bottom);
                    if (fromString == L"Right") return box_value(winrt::CornerRadiusFilterKind::Right);
                    if (fromString == L"ValueOnly") return box_value(winrt::CornerRadiusFilterKind::ValueOnly);
                    throw winrt::hresult_invalid_argument();
                });

            return xamlType;
        });
```

**Comment by jevansaks** [comment]
Oh, shucks. Yeah let's go with what's in master for this release. Can you file a bug on markup, then circle back on the spec review and say we can't do this because of that bug?

**Comment by kaiguo** [comment]
> Oh, shucks. Yeah let's go with what's in master for this release. Can you file a bug on markup, then circle back on the spec review and say we can't do this because of that bug?

Yeah, will do.

**Pull request closed without merging**